enshrine-esdt-safe/src/from_sovereign/transfer_tokens.rs

```diff
@@ -46,23 +42,35 @@ pub trait TransferTokensModule:
             self.emit_transfer_failed_events(
                 &hash_of_hashes,
                 &OperationTuple {
-                    op_hash: operation_hash.clone(),
+                    op_hash: op_hash.clone(),
                     operation: operation.clone(),
                 },
             );
 
             return;
         }
+
+        let token_handler_address = self.token_handler_address().get();
+
+        self.tx()
+            .to(token_handler_address)
+            .typed(token_handler_proxy::TokenHandlerProxy)
+            .mint_tokens(hash_of_hashes, OperationTuple { op_hash, operation })
+            .sync_call();
     }
 
     #[endpoint]
-    fn call_token_handler_mint_endpoint(&self, operation: Operation<Self::Api>) {
+    fn call_token_handler_mint_endpoint(
```

Comment:
What is this endpoint? Why do you need this?
What is `save_minted_tokens` callback at line 74?

Reply:
i used it in some random tests before implementing the distribute payments inside the token-handler contract, should I remove it here or in another PR ?

Reply:
deleted